nuxeo-wopi/src/main/java/org/nuxeo/wopi/jaxrs/WOPIRoot.java

```diff
@@ -52,9 +52,8 @@ public Object filesResource(@PathParam("fileId") FileInfo fileInfo) {
         WebContext context = getContext();
         context.setRepositoryName(fileInfo.repositoryName);
         CoreSession session = context.getCoreSession();
-        // flag the session's principal as a WOPI user for locking policy purpose
-        // TODO find a better way, see NXP-25855
-        LockHelper.markAsWOPIUser(session.getPrincipal());
+        // flag the request as originating from a WOPI client for locking policy purpose
+        LockHelper.setWOPIRequest();
```

Comment:
But we also need to unset it from the thead-local at the end of the request, otherwise another regular request reusing the thread will still think that this is a WOPI request.
You can add a cleanup handler with

    RequestContext.getActiveContext(request).addRequestCleanupHandler()